### PR TITLE
(Minor) More style changes for host filtering functions

### PR DESCRIPTION
**Pull request by jshoe**
- Most of the diff is adding a "result_path" function to do `os.path.join(RESULT_DIR, basename)`
- Test run for this PR: https://staging.idseq.net/samples/622
- Will bump the version number before merging

### CODE REVIEW

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 1]
```diff
@@ -331,97 +339,99 @@ def fetch_genome_work(s3genome, strict):
...
-            # something does not go as planned, because of the prefetching pattern of use.  Also, this function
-            # will fetch whichever of the "tar" or "tar.gz" versions of the archive actually exist in s3,
-            # even if the other version has been specified.  So the two functions are resilient in different ways.
             install_s3mi()
             tarfile = uncompressed(s3genome)
             try:
-                execute_command(
-                    "s3mi cat {tarfile} | tar xvf - -C {refdir}".format(
-                        tarfile=tarfile, refdir=REF_DIR))
+                print("Trying to download compressed genome...")
```
Comment (jameshowardwang) [inline idseq_pipeline/commands/host_filtering_functions.py]: should these be using logger / write_to_log?

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 2]
```diff
@@ -331,97 +339,99 @@ def fetch_genome_work(s3genome, strict):
                            "hg38_pantro5_k16"):
         write_to_log("Oh hello interesting new genome {}".format(genome_name))
     genome_dir = os.path.join(REF_DIR, genome_name)
+
     if not os.path.exists(genome_dir):
         try:
-            #TODO Clean up and fold into fetch_reference
-            # Hmm. I understand the desire to create a single function for fetching any reference we might need,
-            # but the functionality of these two functions is actually quite different.  One of them handles
```
Comment (jameshowardwang) [inline idseq_pipeline/commands/host_filtering_functions.py]: You seem to be deleting these comments without actually implementing what they are suggesting. Is this because we no longer think the recommendation in the TODO is valid?

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 3]
```diff
@@ -583,15 +605,20 @@ def unmapped_files_in(some_dir):
...
         if f is not None:
-            output_i = os.path.join(RESULT_DIR, star_outputs[i])
+            output_i = result_path(star_outputs[i])
             execute_command("mv %s %s;" % (f, output_i))
             uploader_start(output_i, SAMPLE_S3_OUTPUT_PATH + "/")
-    # cleanup
+    # Cleanup
     execute_command("cd %s; rm -rf *" % SCRATCH_DIR)
-    write_to_log("finished job")
+    print("Finished running STAR.")
```
Comment (jameshowardwang) [inline idseq_pipeline/commands/host_filtering_functions.py]: why did you change this?

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 4]
```diff
@@ -613,120 +641,131 @@ def run_priceseqfilter(input_fqs, uploader_start):
...
-        uploader_start(
-            os.path.join(RESULT_DIR, LZW_OUT2), SAMPLE_S3_OUTPUT_PATH + "/")
-    write_to_log("finished job")
+        uploader_start(result_path(LZW_OUT2), dst)
+    print("Finished running LZW.")
 
 
 def run_bowtie2(input_fas, uploader_start):
-    # check if genome downloaded already
+    """Bowtie is an aligner used for filtering out host genomes. Two input
```
Comment (cdebourcy) [inline idseq_pipeline/commands/host_filtering_functions.py]: Love all the clarifying comments regarding external tools. I would change the first sentence of this one to "Bowtie2 is an aligner we use for filtering out reads that map to the host genome"

File: idseq_pipeline/commands/host_filtering_functions.py (hunk 1 above)
Comment (jshoe) [reply idseq_pipeline/commands/host_filtering_functions.py]: I believe they actually do the same thing throughout the program after the logger is configured, but write_to_log is probably better / changed it

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 5]
```diff
@@ -735,12 +774,19 @@ class SkipGsnap(Exception):
 
 
 def run_gsnap_filter(input_fas, uploader_start):
+    """GSNAP is an aligner used for filtering out host genomes. Two input
```
Comment (cdebourcy) [inline idseq_pipeline/commands/host_filtering_functions.py]: Would change to "GSNAP is an aligner we use to filter out reads mapping to the host genome."
Would also add why we do this additional step after STAR/bowtie-based filtering: to increase our sensitivity for host filtering. We currently only do this for the human host -- we use the chimp reference genome as an extension to increase our sensitivity for human matches

File: idseq_pipeline/commands/host_filtering_functions.py (hunk 2 above)
Comment (jshoe) [reply idseq_pipeline/commands/host_filtering_functions.py]: I took this out b/c the "Hmm" part was Boris responding to the TODO (to not combine with fetch_reference), and it seems like they do do pretty different things so I don't think we should combine them. But I could leave it there?

File: idseq_pipeline/commands/host_filtering_functions.py [hunk 6]
```diff
@@ -124,6 +124,10 @@ def fq2fa(input_fastq, output_fasta):
     execute_command(cmd)
 
 
+def result_path(basename):
```
Comment (boris-dimitrov) [inline idseq_pipeline/commands/host_filtering_functions.py]: There is (or was?) a similar result_dir function in non_host_alignment_functions.py, if I recall correctly.

File: idseq_pipeline/commands/host_filtering_functions.py (hunk 2 above)
Comment (boris-dimitrov) [reply idseq_pipeline/commands/host_filtering_functions.py]: Yeah this was a discussion between YF and me in the comments.   I am not sure if we need it or not need it;  but if code cleanliness is an objective, we could record the discussion in an issue and refer to it here, shortening the comment.

File: idseq_pipeline/commands/host_filtering_functions.py (hunk 6 above)
Comment (jshoe) [reply idseq_pipeline/commands/host_filtering_functions.py]: Ah ok I changed result_dir to be called result_path but I think I'll leave the declarations separate since common.py doesn't get the RESULT_DIR (SAMPLE_DIR) declarations

Comment (boris-dimitrov) [reply idseq_pipeline/commands/host_filtering_functions.py]: Sounds good.
